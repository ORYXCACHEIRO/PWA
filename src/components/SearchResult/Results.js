import teste from '../../assets/exterior.jpg';
import { Link } from 'react-router-dom';

import AddFav from '../Hotel/AddFav';
import HotelRating from '../Hotel/HotelRating';

const Result = ({ hoteis, loginStatus }) => {



    if (hoteis.length < 1) {

        return (

            <div>

                <h1 className=' text-4xl font-bold p-3'>No hotel with that name</h1>

            </div>

        )
    }

    return (
        <div className="ml-4 mr-4">
            <div className="inline-flex pt-12 mr-auto ml-auto flex-wrap w-full justify-center gap-16 pb-10 lg:gap-12">

                <div className="ml-4 mr-4">
                    <div className="inline-flex pt-12 mr-auto ml-auto flex-wrap w-full justify-center gap-16 pb-10 lg:gap-12">

                        {hoteis.map((hotel) => (
                            <div key={hotel._id} className="sm:ml-0 ">
                                <div className="bg-purple-500 h-96 w-96 relative sm:right-2 rounded-lg ">
                                    <div className="bg-gray-900 h-96 w-96 absolute left-7 top-7 rounded-lg ">
                                        <div className='relative'>
                                            <div className="top-1 right-[0.15rem] absolute">
                                                <AddFav hotelData={hotel._id} loginStatus={loginStatus} />
                                            </div>
                                            <Link to={"/hotel/" + hotel._id}>
                                                <div className="p-10">
                                                    <img src={teste} alt="" className=" rounded-xl" />
                                                </div>
                                                <div className=' -mt-5 flex gap-5 '>
                                                    <div className='flex  w-max ml-7  items-center'>
                                                        <HotelRating category={hotel.category} />
                                                    </div>
                                                </div>
                                                <div className='mt-2'>
                                                    <h2 className='text-white font-medium text-2xl m-0 ml-7  break-words'>{hotel.name}</h2>
                                                    <h2 className='text-purple-400 font-medium text-lg ml-7 '>{hotel.city}</h2>
                                                </div>
                                            </Link>
                                        </div>
                                    </div>
                                </div>
                            </div>
                        ))}
                    </div>
                </div>
            </div>
        </div>
    );
}

export default Result;