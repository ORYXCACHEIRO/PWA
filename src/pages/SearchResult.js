import Filter from "../components/SearchResult/Filter";
import Result from "../components/SearchResult/Results";
import { useEffect } from "react";
import React, { useState } from "react";


const { search } = window.location;
const query = new URLSearchParams(search).get('s');




const SearchResPage = () => {

    const [userData, setUserData] = useState([])

    useEffect(() => {
        fetch('http://127.0.0.1:5000/search/hotel/' + query , {
            method: 'GET',
            headers: { 'Accept': 'application/json' }
        })
        .then(response => response.json())
        .then(data => setUserData(data))

    }, []);



return (
    <>
        <div className="mt-10">
            <h1 className="font-bold text-5xl text-center">Search result with <span className="text-purple-500">{query}</span>.</h1>
        </div>
        <h1></h1>
        <div className="pt-10 flex md:flex-col">
            <div className=" w-1/4 xl:w-max md:w-full">
                <Filter />
            </div>
            <div className=" w-3/4 md:w-full">
                <Result hoteis={userData}  />
            </div>
        </div>
    </>
)

}

export default SearchResPage;