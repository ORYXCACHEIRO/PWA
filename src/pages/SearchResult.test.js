import React from 'react'
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import SearchResult from './SearchResult';
import renderer from 'react-test-renderer';


Enzyme.configure({ adapter: new Adapter() });

describe("SearchResult", () => {


        it('Should render correctly', function () {
            const component = renderer.create(<SearchResult />);
            const tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });

})