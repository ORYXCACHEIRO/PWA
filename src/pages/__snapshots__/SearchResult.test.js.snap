// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult Should render correctly 1`] = `
Array [
  <div
    className="mt-10"
  >
    <h1
      className="font-bold text-5xl text-center"
    >
      Search result with 
      <span
        className="text-purple-500"
      />
      .
    </h1>
  </div>,
  <div
    className="pt-10 flex md:flex-col"
  >
    <div
      className=" w-1/4 xl:w-max md:w-full"
    >
      <div
        className="bg-gray-900 w-80 rounded-xl ml-auto mr-auto flex flex-col mt-10 md:w-3/4 md:mt-5 xl:ml-5 md:ml-auto"
      >
        <div
          className="p-3"
        >
          <div
            className="m-2"
          >
            <h1
              className="text-3xl font-bold pl-3 text-white border-b-white border-b-3 pb-2 "
            >
              Filter by :
            </h1>
          </div>
        </div>
        <div
          className="p-3 flex flex-col"
        >
          <div
            className=" m-2 text-white border-white border-3  rounded-lg "
          >
            <h1
              className="text-2xl font-bold pl-3 text-white2 m-2 "
            >
              Name
            </h1>
            <div
              className="border-t-white border-t-2 flex  gap-3 p-3 flex-col"
            >
              <div
                className="flex items-center gap-3"
              >
                <input
                  className="appearance-none ml-2 h-5 w-5 bg-white rounded-md checked:border-2 checked:border-purple-500 checked:bg-check checked:bg-purple-500 "
                  type="checkbox"
                />
                <span
                  className="flex items-center gap-2 text-xl"
                >
                  Order 
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z"
                    />
                  </svg>
                </span>
              </div>
              <div
                className="flex items-center gap-3"
              >
                <input
                  className="appearance-none ml-2 h-5 w-5 bg-white rounded-md checked:border-2 checked:border-purple-500 checked:bg-check checked:bg-purple-500 "
                  type="checkbox"
                />
                <span
                  className="flex items-center gap-2 text-xl"
                >
                  Order 
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z"
                    />
                  </svg>
                </span>
              </div>
            </div>
          </div>
          <div
            className=" m-2 text-white border-white border-3  rounded-lg "
          >
            <h1
              className="text-2xl font-bold pl-3 text-white2 m-2 "
            >
              Comodities
            </h1>
            <div
              className="border-t-white border-t-2 flex  gap-3 p-3 flex-col overflow-y-auto h-32 scrollbar-thin scrollbar-thumb-purple-500 scrollbar-track-gray-100"
            >
              <div
                className="flex items-center gap-3"
              >
                <input
                  className="appearance-none ml-2 h-5 w-5 bg-white rounded-md checked:border-2 checked:border-purple-500 checked:bg-check checked:bg-purple-500 "
                  type="checkbox"
                />
                <span
                  className="flex items-center gap-2 text-xl"
                >
                  Order 
                </span>
              </div>
              <div
                className="flex items-center gap-3"
              >
                <input
                  className="appearance-none ml-2 h-5 w-5 bg-white rounded-md checked:border-2 checked:border-purple-500 checked:bg-check checked:bg-purple-500 "
                  type="checkbox"
                />
                <span
                  className="flex items-center gap-2 text-xl"
                >
                  Order 
                </span>
              </div>
              <div
                className="flex items-center gap-3"
              >
                <input
                  className="appearance-none ml-2 h-5 w-5 bg-white rounded-md checked:border-2 checked:border-purple-500 checked:bg-check checked:bg-purple-500 "
                  type="checkbox"
                />
                <span
                  className="flex items-center gap-2 text-xl"
                >
                  Order 
                </span>
              </div>
              <div
                className="flex items-center gap-3"
              >
                <input
                  className="appearance-none ml-2 h-5 w-5 bg-white rounded-md checked:border-2 checked:border-purple-500 checked:bg-check checked:bg-purple-500 "
                  type="checkbox"
                />
                <span
                  className="flex items-center gap-2 text-xl"
                >
                  Order 
                </span>
              </div>
              <div
                className="flex items-center gap-3"
              >
                <input
                  className="appearance-none ml-2 h-5 w-5 bg-white rounded-md checked:border-2 checked:border-purple-500 checked:bg-check checked:bg-purple-500 "
                  type="checkbox"
                />
                <span
                  className="flex items-center gap-2 text-xl"
                >
                  Order
                </span>
              </div>
            </div>
          </div>
          <div
            className=" m-2 text-white border-white border-3  rounded-lg "
          >
            <h1
              className="text-2xl font-bold pl-3 text-white2 m-2 "
            >
              Languages
            </h1>
            <div
              className="border-t-white border-t-2 flex  gap-3 p-3 flex-col overflow-y-auto h-32 scrollbar-thin scrollbar-thumb-purple-500 scrollbar-track-gray-100"
            >
              <div
                className="flex items-center gap-3"
              >
                <input
                  className="appearance-none ml-2 h-5 w-5 bg-white rounded-md checked:border-2 checked:border-purple-500 checked:bg-check checked:bg-purple-500 "
                  type="checkbox"
                />
                <span
                  className="flex items-center gap-2 text-xl"
                >
                  Order 
                </span>
              </div>
              <div
                className="flex items-center gap-3"
              >
                <input
                  className="appearance-none ml-2 h-5 w-5 bg-white rounded-md checked:border-2 checked:border-purple-500 checked:bg-check checked:bg-purple-500 "
                  type="checkbox"
                />
                <span
                  className="flex items-center gap-2 text-xl"
                >
                  Order 
                </span>
              </div>
              <div
                className="flex items-center gap-3"
              >
                <input
                  className="appearance-none ml-2 h-5 w-5 bg-white rounded-md checked:border-2 checked:border-purple-500 checked:bg-check checked:bg-purple-500 "
                  type="checkbox"
                />
                <span
                  className="flex items-center gap-2 text-xl"
                >
                  Order 
                </span>
              </div>
              <div
                className="flex items-center gap-3"
              >
                <input
                  className="appearance-none ml-2 h-5 w-5 bg-white rounded-md checked:border-2 checked:border-purple-500 checked:bg-check checked:bg-purple-500 "
                  type="checkbox"
                />
                <span
                  className="flex items-center gap-2 text-xl"
                >
                  Order 
                </span>
              </div>
              <div
                className="flex items-center gap-3"
              >
                <input
                  className="appearance-none ml-2 h-5 w-5 bg-white rounded-md checked:border-2 checked:border-purple-500 checked:bg-check checked:bg-purple-500 "
                  type="checkbox"
                />
                <span
                  className="flex items-center gap-2 text-xl"
                >
                  Order
                </span>
              </div>
            </div>
          </div>
          <div
            className=" m-2 text-white border-white border-3  rounded-lg "
          >
            <h1
              className="text-2xl font-bold pl-3 text-white2 m-2 "
            >
              Rating
            </h1>
            <div
              className="border-t-white border-t-2 flex  gap-3 p-3 flex-col overflow-y-auto h-32 scrollbar-thin scrollbar-thumb-purple-500 scrollbar-track-gray-100"
            >
              <div
                className="flex items-center gap-3"
              >
                <input
                  className="appearance-none ml-2 h-5 w-5 bg-white rounded-md checked:border-2 checked:border-purple-500 checked:bg-check checked:bg-purple-500 "
                  type="checkbox"
                />
                <span
                  className="flex items-center gap-2 text-xl"
                >
                  Order 
                </span>
              </div>
              <div
                className="flex items-center gap-3"
              >
                <input
                  className="appearance-none ml-2 h-5 w-5 bg-white rounded-md checked:border-2 checked:border-purple-500 checked:bg-check checked:bg-purple-500 "
                  type="checkbox"
                />
                <span
                  className="flex items-center gap-2 text-xl"
                >
                  Order 
                </span>
              </div>
              <div
                className="flex items-center gap-3"
              >
                <input
                  className="appearance-none ml-2 h-5 w-5 bg-white rounded-md checked:border-2 checked:border-purple-500 checked:bg-check checked:bg-purple-500 "
                  type="checkbox"
                />
                <span
                  className="flex items-center gap-2 text-xl"
                >
                  Order 
                </span>
              </div>
              <div
                className="flex items-center gap-3"
              >
                <input
                  className="appearance-none ml-2 h-5 w-5 bg-white rounded-md checked:border-2 checked:border-purple-500 checked:bg-check checked:bg-purple-500 "
                  type="checkbox"
                />
                <span
                  className="flex items-center gap-2 text-xl"
                >
                  Order 
                </span>
              </div>
              <div
                className="flex items-center gap-3"
              >
                <input
                  className="appearance-none ml-2 h-5 w-5 bg-white rounded-md checked:border-2 checked:border-purple-500 checked:bg-check checked:bg-purple-500 "
                  type="checkbox"
                />
                <span
                  className="flex items-center gap-2 text-xl"
                >
                  Order
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className=" w-3/4 md:w-full"
    >
      <div>
        <h1
          className=" text-4xl font-bold p-3"
        >
          No hotel with that name
        </h1>
      </div>
    </div>
  </div>,
]
`;
